### PR TITLE
Parse Microdata to discover page config

### PR DESCRIPTION
**Pull request by sohanirao**

### CODE REVIEW

File: src/model/page-config-resolver-test.js [hunk 1]
```diff
@@ -266,7 +270,143 @@ describes.realWin('PageConfigResolver', {}, env => {
...
+      readyState = 'loading';
+      Object.defineProperty(doc, 'readyState', {get: () => readyState});
+    });
+
+    function getTestDiv(options) {
+      const access = (options['isAccessibleForFree'] != undefined);
+      const productId = options['productId'];
+      const topItemtype = options['newsArticleType'];
+      const divElement = createElement(doc, 'div');
+      if (!!topItemtype) {
```
Comment (dvoytenko) [inline src/model/page-config-resolver-test.js]: Just `if (topItemType) {}`. Here and below.

File: src/model/page-config-resolver.js [hunk 2]
```diff
@@ -291,6 +358,64 @@ class MicrodataParser {
     this.win_ = win;
   }
 
+  discoverAccess_(prop) {
+    if (prop.getAttribute('itemprop') == 'isAccessibleForFree') {
+      const value = prop.content || prop.textContent;
+      if (typeof value == 'boolean') {
```
Comment (dvoytenko) [inline src/model/page-config-resolver.js]: Can it ever really be a `boolean`?

File: src/model/page-config-resolver.js [hunk 3]
```diff
@@ -291,6 +358,64 @@ class MicrodataParser {
...
+    }
+    return null;
+  }
+
+  discoverProductId_(prop) {
+    if (prop.getAttribute('itemprop') == 'productID')	{
+      return prop.content || prop.textContent;
+    }
+    return null;
+  };
```
Comment (dvoytenko) [inline src/model/page-config-resolver.js]: no semicolon

File: src/model/page-config-resolver.js [hunk 4]
```diff
@@ -291,6 +358,64 @@ class MicrodataParser {
...
+          if (element.getAttribute('itemtype') == 'http://schema.org/NewsArticle') {
+            const props = element.querySelectorAll('[itemprop]');
+            const items = [];
+            props.forEach(prop => {
+              let type = 'default';
+              if (prop.matches('[itemscope]')) {
+                type = prop.getAttribute('itemtype');
+              }
+              const item = {
+                'type': type,
```
Comment (dvoytenko) [inline src/model/page-config-resolver.js]: Why are these properties in quotes?

File: src/model/page-config-resolver.js [hunk 5]
```diff
@@ -291,6 +358,64 @@ class MicrodataParser {
...
+    }
+    return null;
+  };
+
+  tryExtractConfig_(doc) {
+    let itemId = 0;
+    const results = [];
+    doc.querySelectorAll('[itemscope]')
+        .forEach(element => {
+          if (element.getAttribute('itemtype') == 'http://schema.org/NewsArticle') {
```
Comment (dvoytenko) [inline src/model/page-config-resolver.js]: Can we combine `itemtype` with selector above? That'd be a lot faster.

File: src/model/page-config-resolver.js (hunk 2 above)
Comment (dvoytenko) [reply src/model/page-config-resolver.js]: JSON-LD is in the JSON format that has a native support for booleans. But microdata is parsed from attributes/text content, so my question was how is boolean possible there?

File: src/model/page-config-resolver.js (hunk 5 above)
Comment (sohanirao) [reply src/model/page-config-resolver.js]: What if itemscope is defined in the parent div and itemtype is defined as a meta of a child div? We will end up missing it.

Comment (dvoytenko) [reply src/model/page-config-resolver.js]: But that's not what you're doing now, is it?

**Pull request closed without merging**